app/controllers/api/v1/orders_controller.rb

```diff
@@ -133,6 +133,15 @@ def serializer
         Api::V1::OrderSerializer
       end
 
+      def shallow_serializer
+        Api::V1::OrderShallowSerializer
+      end
+
+      def select_serializer
+        is_shallow_render = params[:shallow] == true || params[:shallow] == 'true'
```

Comment:
Rails should ensure all params are strings. So no need to test for the true bool here. For the rspec test below, send the string 'true' and all should pass.